### PR TITLE
feat(#4): Open Source - release

### LINKED ISSUE #4: Open Source - release

### PR DESCRIPTION
**Pull request by Alberdi**
* Added Apache LICENSE, README and CONTRIBUTING.
* Updated the tests to 100% coverage and added xref and dialyzer checks.
* Removed unused nbully_debug module.
* Other minor esthetic changes.

Closes #4.

### CODE REVIEW

File: src/nbully.app.src [hunk 1]
```diff
@@ -1,7 +1,7 @@
 %%% -*- mode:erlang -*-
```
Comment (javiergarea) [inline src/nbully.app.src:1]: Should we keep these headers?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -7,14 +7,15 @@ on:
     branches: [main]
 
 env:
-  REBAR3-VERSION: 3.20.0
+  OTP-VERSION: 26.2.1
```
Comment (javiergarea) [inline .github/workflows/ci.yml]: With the strategy matrix this should be no longer necessary.

**Review by javiergarea — APPROVED** [review]
LGTM 🚀 
Nitpick: I would not enforce coverage at 100%, but won't block open source releases becase of it 😆

File: src/nbully.app.src (hunk 1 above)
Comment (Alberdi) [reply src/nbully.app.src:1]: I will remove them!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Alberdi) [reply .github/workflows/ci.yml]: Sure! I will remove it.

**Comment by Alberdi** [comment]
I downgraded the coverage expectation to 95% because in Erlang 25 the `peer` module doesn't mark as covered some of the calls done by the new nodes (maybe due to https://github.com/erlang/otp/commit/d7f1801788c7c4af01498e991641d3c47125ba7f).

There are still some (previous) flakiness issues with the tests, but I will try to review them at a later date.

**Review by mrrubinos — APPROVED** [review]

**Pull request merged**